core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java

```diff
@@ -111,6 +118,7 @@ private void removeIgnored(Map<DotName, Set<String>> unindexedClasses,
     }
 
     private void addReflectiveHierarchy(CombinedIndexBuildItem combinedIndexBuildItem,
+            List<GeneratedClassesIndexBuildItem> generatedClassesIndexBuildItem,
```

Comment:
Do we really expect a `List`?

Reply:
yes, as many extensions can contribute their selected classes which could not be beans etc.